### PR TITLE
Post Settings NPEs

### PR DESCRIPTION
**Pull request by oguzkocer**
This is a much better fix for #6319. The issue we were having is that when the activity was getting re-created, the fragment was calling `onCreate` before the activity calls its. Moving the initialization to `onActivityCreated` fixes most of the issues, but NPEs could still happen in callbacks. So, a few additional checks has been added to handle edge cases. Ignoring some callbacks is not great, but I think this is the best we can do right now and it should only happen if a callback happens while activity is being re-created.

To test:
* Enable "Do not keep activities" in settings
* Go into Post Settings, go to bg and re-open the app. (The keyboard will open automatically as a result and this will be handled separately in #6361.

/cc @maxme

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java [hunk 1]
```diff
@@ -616,10 +616,18 @@ public void onTimeSet(TimePicker timePicker, int selectedHour, int selectedMinut
     // Helpers
 
     private PostModel getPost() {
+        if (mEditPostActivityHook == null) {
+            // This can only happen during a callback while activity is re-created for some reason (config changes etc)
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: 👍
